akka-persistence-testkit/src/main/scala/akka/persistence/testkit/javadsl/SnapshotTestKit.scala

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.javadsl
+
+import akka.actor.ActorSystem
+import akka.persistence.testkit.scaladsl.{ SnapshotTestKit => ScalaTestKit }
+import akka.persistence.testkit.{ ExpectedFailure, SnapshotMeta, SnapshotOperation, SnapshotStorage }
+import akka.util.JavaDurationConverters._
+import akka.util.ccompat.JavaConverters._
+import java.time.Duration
+import java.util.{ List => JList }
+import java.util.{ function => jf }
+
+import akka.annotation.ApiMayChange
+import akka.japi.Pair
+import akka.testkit.javadsl.CachingPartialFunction
+
+/**
+ * Class for testing persisted snapshots in persistent actors.
+ */
+@ApiMayChange
+class SnapshotTestKit(system: ActorSystem) {
+  import akka.persistence.testkit.internal.Utils.JavaCollectionConversions._
+  import akka.persistence.testkit.internal.Utils.JavaFuncConversions._
+
+  private val scalaTestkit = new ScalaTestKit(system)
+
+  /**
+   * Check that nothing has been saved in the storage.
+   */
+  def expectNothingPersisted(persistenceId: String): Unit = scalaTestkit.expectNothingPersisted(persistenceId)
+
+  /**
+   * Check for `max` time that nothing has been saved in the storage.
+   */
+  def expectNothingPersisted(persistenceId: String, max: Duration): Unit =
+    scalaTestkit.expectNothingPersisted(persistenceId, max.asScala)
+
+  /**
+   * Check that `msg` event has been saved in the storage.
+   */
+  def expectNextPersisted[A](persistenceId: String, msg: A): A = scalaTestkit.expectNextPersisted(persistenceId, msg)
+
+  /**
+   * Check for `max` time that `msg` event has been saved in the storage.
+   */
+  def expectNextPersisted[A](persistenceId: String, msg: A, max: Duration): A =
+    scalaTestkit.expectNextPersisted(persistenceId, msg, max.asScala)
+
+  /**
+   * Check that next persisted in storage for particular persistence id event matches partial function `pf`.
+   */
+  def expectNextPersistedPF[A](persistenceId: String, pf: jf.Function[Any, A]): A =
```

Comment:
Partial function for Java doesn't work well.

In Typed TestKit we have moved away from partial function asserts, also in scaladsl, because the error messages are not good. Instead we have the exact match with `expectMessage` and otherwise `expectMessageType` that can then be followed by more specific asserts using ScalaTest/JUnit.

Would be good to align this with Typed TestKit. See `akka.actor.testkit.typed.scaladsl.TestProbe` and `akka.actor.testkit.typed.javadsl.TestProbe`.

In other words, remove the PF methods and add `expectNextPersistedClass` in javadsl and `expectNextPersistedType` in scaladsl.